The job **`x86_64-gnu-llvm-18`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/13525457481/job/37794785419) [(plain)](https://github.com/rust-lang/rust/commit/819c62630b180ecd6877bedbf2725a0678dece23/checks/37794785419/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
#22 exporting to docker image format
#22 sending tarball 28.0s done
#22 DONE 33.8s
##[endgroup]
Setting extra environment values for docker:  --env ENABLE_GCC_CODEGEN=1 --env GCC_EXEC_PREFIX=/usr/lib/gcc/
[CI_JOB_NAME=x86_64-gnu-llvm-18]
debug: `DISABLE_CI_RUSTC_IF_INCOMPATIBLE` configured.
---
sccache: Listening on address 127.0.0.1:4226
##[group]Configure the build
configure: processing command line
configure: 
configure: build.configure-args := ['--build=x86_64-unknown-linux-gnu', '--llvm-root=/usr/lib/llvm-18', '--enable-llvm-link-shared', '--set', 'rust.randomize-layout=true', '--set', 'rust.thin-lto-import-instr-limit=10', '--enable-verbose-configure', '--enable-sccache', '--disable-manage-submodules', '--enable-locked-deps', '--enable-cargo-native-static', '--set', 'rust.codegen-units-std=1', '--set', 'dist.compression-profile=balanced', '--dist-compression-formats=xz', '--set', 'rust.lld=false', '--disable-dist-src', '--release-channel=nightly', '--enable-debug-assertions', '--enable-overflow-checks', '--enable-llvm-assertions', '--set', 'rust.verify-llvm-ir', '--set', 'rust.codegen-backends=llvm,cranelift,gcc', '--set', 'llvm.static-libstdcpp', '--enable-new-symbol-mangling']
configure: target.x86_64-unknown-linux-gnu.llvm-config := /usr/lib/llvm-18/bin/llvm-config
configure: llvm.link-shared     := True
configure: rust.randomize-layout := True
configure: rust.thin-lto-import-instr-limit := 10
---

1 error: attribute value must be a literal
2   --> $DIR/empty-expansion-macro.rs:1:1
3    |
- LL | #[crate_type = foo!()]
+ LL | / // Tests for the issue in #137589
+ ...  |
+ ...  |
+ LL | | #[crate_type = foo!()]
6 
7 error: unexpected end of macro invocation
-   --> $DIR/empty-expansion-macro.rs:4:1
+   --> $DIR/empty-expansion-macro.rs:7:1
---
13 error: cannot find macro `foo` in this scope
-   --> $DIR/empty-expansion-macro.rs:1:16
+   --> $DIR/empty-expansion-macro.rs:4:16
15    |
16 LL | #[crate_type = foo!()]
17    |                ^^^ consider moving the definition of `foo` before this call
18    |
19 note: a macro with the same name exists, but it appears later
-   --> $DIR/empty-expansion-macro.rs:4:14
+   --> $DIR/empty-expansion-macro.rs:7:14
+   --> $DIR/empty-expansion-macro.rs:7:14
21    |
22 LL | macro_rules! foo {}
23    |              ^^^

Note: some mismatched output was normalized before being compared
- LL | | #[crate_type = foo!()] //~ ERROR attribute value must be a literal
-   --> /checkout/tests/ui/attributes/empty-expansion-macro.rs:4:16
-   --> /checkout/tests/ui/attributes/empty-expansion-macro.rs:7:14
+ LL | / // Tests for the issue in #137589
+ ...  |
+ ...  |
+ LL | | #[crate_type = foo!()]
+   --> $DIR/empty-expansion-macro.rs:7:1
+   --> $DIR/empty-expansion-macro.rs:4:16
+   --> $DIR/empty-expansion-macro.rs:7:14



The actual stderr differed from the expected stderr.
To update references, rerun the tests and pass the `--bless` flag
To only update this specific test, also pass `--test-args attributes/empty-expansion-macro.rs`

error: 1 errors occurred comparing output.
status: exit status: 1
command: env -u RUSTC_LOG_COLOR RUSTC_ICE="0" RUST_BACKTRACE="short" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/tests/ui/attributes/empty-expansion-macro.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "-Z" "ignore-directory-in-diagnostics-source-blocks=/cargo" "-Z" "ignore-directory-in-diagnostics-source-blocks=/checkout/vendor" "--sysroot" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2" "--target=x86_64-unknown-linux-gnu" "--check-cfg" "cfg(test,FALSE)" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Zwrite-long-types-to-disk=no" "-Cstrip=debuginfo" "--emit" "metadata" "-C" "prefer-dynamic" "--out-dir" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/attributes/empty-expansion-macro" "-A" "unused" "-A" "internal_features" "-Crpath" "-Cdebuginfo=0" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers"
--- stderr -------------------------------
error: attribute value must be a literal
##[error]  --> /checkout/tests/ui/attributes/empty-expansion-macro.rs:1:1
   |
   |
LL | / // Tests for the issue in #137589
...  |
LL | | #[crate_type = foo!()] //~ ERROR attribute value must be a literal

error: unexpected end of macro invocation
##[error]  --> /checkout/tests/ui/attributes/empty-expansion-macro.rs:7:1
   |
   |
LL | macro_rules! foo {} //~ ERROR unexpected end of macro invocation
   | ^^^^^^^^^^^^^^^^^^^ missing tokens in macro arguments
error: cannot find macro `foo` in this scope
##[error]  --> /checkout/tests/ui/attributes/empty-expansion-macro.rs:4:16
   |
   |
LL | #[crate_type = foo!()] //~ ERROR attribute value must be a literal
   |                ^^^ consider moving the definition of `foo` before this call
note: a macro with the same name exists, but it appears later
  --> /checkout/tests/ui/attributes/empty-expansion-macro.rs:7:14
   |
   |
LL | macro_rules! foo {} //~ ERROR unexpected end of macro invocation

error: aborting due to 3 previous errors
------------------------------------------



---- [ui] tests/ui/attributes/key-value-expansion-on-mac.rs stdout ----
Saved the actual stderr to "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/attributes/key-value-expansion-on-mac/key-value-expansion-on-mac.stderr"
diff of stderr:

4 LL | #[rustc_dummy = stringify!(b)]
6 
- error: aborting due to 1 previous error
+ error: attribute value must be a literal
+   --> $DIR/key-value-expansion-on-mac.rs:1:1
+   --> $DIR/key-value-expansion-on-mac.rs:1:1
+    |
+ LL | / #![feature(rustc_attrs)]
+ LL | |
+ LL | | #[rustc_dummy = stringify!(a)] // OK
+ 
+ error: aborting due to 2 previous errors
8 
9 
9 

Note: some mismatched output was normalized before being compared
-   --> /checkout/tests/ui/attributes/key-value-expansion-on-mac.rs:1:1
+ error: attribute value must be a literal
+   --> $DIR/key-value-expansion-on-mac.rs:1:1
+    |
+ LL | / #![feature(rustc_attrs)]
+ LL | |
+ LL | | #[rustc_dummy = stringify!(a)] // OK
+ 
+ error: aborting due to 2 previous errors



The actual stderr differed from the expected stderr.
To update references, rerun the tests and pass the `--bless` flag
To only update this specific test, also pass `--test-args attributes/key-value-expansion-on-mac.rs`

error: 1 errors occurred comparing output.
status: exit status: 1
command: env -u RUSTC_LOG_COLOR RUSTC_ICE="0" RUST_BACKTRACE="short" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/tests/ui/attributes/key-value-expansion-on-mac.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "-Z" "ignore-directory-in-diagnostics-source-blocks=/cargo" "-Z" "ignore-directory-in-diagnostics-source-blocks=/checkout/vendor" "--sysroot" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2" "--target=x86_64-unknown-linux-gnu" "--check-cfg" "cfg(test,FALSE)" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Zwrite-long-types-to-disk=no" "-Cstrip=debuginfo" "--emit" "metadata" "-C" "prefer-dynamic" "--out-dir" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/attributes/key-value-expansion-on-mac" "-A" "unused" "-A" "internal_features" "-Crpath" "-Cdebuginfo=0" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers"
--- stderr -------------------------------
error: attribute value must be a literal
##[error]  --> /checkout/tests/ui/attributes/key-value-expansion-on-mac.rs:11:17
   |
   |
LL | #[rustc_dummy = stringify!(b)] //~ ERROR attribute value must be a literal

error: attribute value must be a literal
##[error]  --> /checkout/tests/ui/attributes/key-value-expansion-on-mac.rs:1:1
   |
   |
LL | / #![feature(rustc_attrs)]
LL | |
LL | | #[rustc_dummy = stringify!(a)] // OK

error: aborting due to 2 previous errors
------------------------------------------



---- [ui] tests/ui/attributes/issue-90873.rs stdout ----
Saved the actual stderr to "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/attributes/issue-90873/issue-90873.stderr"
diff of stderr:

22 LL | #![a={impl std::ops::Neg for i8 {}}]
24 
+ error: attribute value must be a literal
+   --> $DIR/issue-90873.rs:1:5
+    |
+    |
+ LL | #![u=||{static d=||1;}]
+ 
+ error: attribute value must be a literal
+   --> $DIR/issue-90873.rs:6:5
+    |
+    |
+ LL | #![a={impl std::ops::Neg for i8 {}}]
+ 
25 error[E0601]: `main` function not found in crate `issue_90873`
26   --> $DIR/issue-90873.rs:6:37
27    |
27    |

39 LL | #![u=||{static d: <type>=||1;}]
41 
- error: aborting due to 6 previous errors
+ error: aborting due to 8 previous errors
43 
---
-   --> /checkout/tests/ui/attributes/issue-90873.rs:6:5
+ error: attribute value must be a literal
+   --> $DIR/issue-90873.rs:1:5
+    |
+ LL | #![u=||{static d=||1;}]
+ 
+ error: attribute value must be a literal
+   --> $DIR/issue-90873.rs:6:5
+    |
+    |
+ LL | #![a={impl std::ops::Neg for i8 {}}]
+ 
+ error: aborting due to 8 previous errors



The actual stderr differed from the expected stderr.
To update references, rerun the tests and pass the `--bless` flag
To only update this specific test, also pass `--test-args attributes/issue-90873.rs`

error: 1 errors occurred comparing output.
status: exit status: 1
command: env -u RUSTC_LOG_COLOR RUSTC_ICE="0" RUST_BACKTRACE="short" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/tests/ui/attributes/issue-90873.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "-Z" "ignore-directory-in-diagnostics-source-blocks=/cargo" "-Z" "ignore-directory-in-diagnostics-source-blocks=/checkout/vendor" "--sysroot" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2" "--target=x86_64-unknown-linux-gnu" "--check-cfg" "cfg(test,FALSE)" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Zwrite-long-types-to-disk=no" "-Cstrip=debuginfo" "--emit" "metadata" "-C" "prefer-dynamic" "--out-dir" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/attributes/issue-90873" "-A" "unused" "-A" "internal_features" "-Crpath" "-Cdebuginfo=0" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers"
--- stderr -------------------------------
error: attribute value must be a literal
##[error]  --> /checkout/tests/ui/attributes/issue-90873.rs:1:6
   |
   |
LL | #![u=||{static d=||1;}]

error: attribute value must be a literal
##[error]  --> /checkout/tests/ui/attributes/issue-90873.rs:6:6
   |
   |
LL | #![a={impl std::ops::Neg for i8 {}}]

error: cannot find attribute `u` in this scope
##[error]  --> /checkout/tests/ui/attributes/issue-90873.rs:1:4
   |
   |
LL | #![u=||{static d=||1;}]

error: cannot find attribute `a` in this scope
##[error]  --> /checkout/tests/ui/attributes/issue-90873.rs:6:4
   |
   |
LL | #![a={impl std::ops::Neg for i8 {}}]

error: attribute value must be a literal
##[error]  --> /checkout/tests/ui/attributes/issue-90873.rs:1:5
   |
   |
LL | #![u=||{static d=||1;}]

error: attribute value must be a literal
##[error]  --> /checkout/tests/ui/attributes/issue-90873.rs:6:5
   |
   |
LL | #![a={impl std::ops::Neg for i8 {}}]

error[E0601]: `main` function not found in crate `issue_90873`
##[error]  --> /checkout/tests/ui/attributes/issue-90873.rs:6:37
   |
   |
LL | #![a={impl std::ops::Neg for i8 {}}]
   |                                     ^ consider adding a `main` function to `/checkout/tests/ui/attributes/issue-90873.rs`
error: missing type for `static` item
##[error]  --> /checkout/tests/ui/attributes/issue-90873.rs:1:17
   |
   |
LL | #![u=||{static d=||1;}]
   |
help: provide a type for the item
   |
   |
LL | #![u=||{static d: <type>=||1;}]

error: aborting due to 8 previous errors

For more information about this error, try `rustc --explain E0601`.
For more information about this error, try `rustc --explain E0601`.
------------------------------------------


---- [ui] tests/ui/attributes/unused-item-in-attr.rs stdout ----
Saved the actual stderr to "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/attributes/unused-item-in-attr/unused-item-in-attr.stderr"
diff of stderr:

10 LL | #[w = { extern crate alloc; }]
12 
- error: aborting due to 2 previous errors
+ error: attribute value must be a literal
+   --> $DIR/unused-item-in-attr.rs:1:5
+   --> $DIR/unused-item-in-attr.rs:1:5
+    |
+ LL | #[w = { extern crate alloc; }]
+ 
+ error: aborting due to 3 previous errors
14 
15 
15 

Note: some mismatched output was normalized before being compared
-   --> /checkout/tests/ui/attributes/unused-item-in-attr.rs:1:5
+ error: attribute value must be a literal
+   --> $DIR/unused-item-in-attr.rs:1:5
+    |
+ LL | #[w = { extern crate alloc; }]
+ 
+ error: aborting due to 3 previous errors



The actual stderr differed from the expected stderr.
To update references, rerun the tests and pass the `--bless` flag
To only update this specific test, also pass `--test-args attributes/unused-item-in-attr.rs`

error: 1 errors occurred comparing output.
status: exit status: 1
command: env -u RUSTC_LOG_COLOR RUSTC_ICE="0" RUST_BACKTRACE="short" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/tests/ui/attributes/unused-item-in-attr.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "-Z" "ignore-directory-in-diagnostics-source-blocks=/cargo" "-Z" "ignore-directory-in-diagnostics-source-blocks=/checkout/vendor" "--sysroot" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2" "--target=x86_64-unknown-linux-gnu" "--check-cfg" "cfg(test,FALSE)" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Zwrite-long-types-to-disk=no" "-Cstrip=debuginfo" "--emit" "metadata" "-C" "prefer-dynamic" "--out-dir" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/attributes/unused-item-in-attr" "-A" "unused" "-A" "internal_features" "-Crpath" "-Cdebuginfo=0" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers"
--- stderr -------------------------------
error: attribute value must be a literal
##[error]  --> /checkout/tests/ui/attributes/unused-item-in-attr.rs:1:7
   |
   |
LL | #[w = { extern crate alloc; }]

error: cannot find attribute `w` in this scope
##[error]  --> /checkout/tests/ui/attributes/unused-item-in-attr.rs:1:3
   |
   |
LL | #[w = { extern crate alloc; }]

error: attribute value must be a literal
##[error]  --> /checkout/tests/ui/attributes/unused-item-in-attr.rs:1:5
   |
   |
LL | #[w = { extern crate alloc; }]

error: aborting due to 3 previous errors
------------------------------------------

---
- error: aborting due to 3 previous errors
+ error: attribute value must be a literal
+   --> $DIR/key-value-expansion.rs:14:23
+    |
+ LL |         #[rustc_dummy = $expr] // Any key-value attribute, not necessarily `doc`
+ ...
+ ...
+ LL | bug!((column!()));
+    |
+    = note: this error originates in the macro `bug` (in Nightly builds, run with -Z macro-backtrace for more info)
+ 
+ error: attribute value must be a literal
+ error: attribute value must be a literal
+   --> $DIR/key-value-expansion.rs:30:15
+    |
+ LL |         #[doc = $test]
+ ...
+ ...
+ LL | bug!();
+    |
+    = note: this error originates in the macro `bug` (in Nightly builds, run with -Z macro-backtrace for more info)
+ 
+ error: attribute value must be a literal
+ error: attribute value must be a literal
+   --> $DIR/key-value-expansion.rs:41:15
+    |
+ LL |         #[doc = $x]
+ ...
+ ...
+ LL | some_macro!(u8);
+    |
+    = note: this error originates in the macro `doc_comment` which comes from the expansion of the macro `some_macro` (in Nightly builds, run with -Z macro-backtrace for more info)
+ 
+ error: aborting due to 6 previous errors
+ error: aborting due to 6 previous errors
30 
31 

Note: some mismatched output was normalized before being compared
-   --> /checkout/tests/ui/attributes/key-value-expansion.rs:14:23
- LL | bug!((column!())); //~ ERROR attribute value must be a literal
-   --> /checkout/tests/ui/attributes/key-value-expansion.rs:41:15
+ error: attribute value must be a literal
+   --> $DIR/key-value-expansion.rs:14:23
+    |
+    |
+ LL |         #[rustc_dummy = $expr] // Any key-value attribute, not necessarily `doc`
+ ...
+ ...
+ LL | bug!((column!()));
+    |
+    = note: this error originates in the macro `bug` (in Nightly builds, run with -Z macro-backtrace for more info)
+ 
+ error: attribute value must be a literal
+ error: attribute value must be a literal
+   --> $DIR/key-value-expansion.rs:30:15
+    |
+ LL |         #[doc = $test]
+ ...
+ ...
+ LL | bug!();
+    |
+    = note: this error originates in the macro `bug` (in Nightly builds, run with -Z macro-backtrace for more info)
+ 
+ error: attribute value must be a literal
+ error: attribute value must be a literal
+   --> $DIR/key-value-expansion.rs:41:15
+    |
+ LL |         #[doc = $x]
+ ...
+ ...
+ LL | some_macro!(u8);
+    |
+    = note: this error originates in the macro `doc_comment` which comes from the expansion of the macro `some_macro` (in Nightly builds, run with -Z macro-backtrace for more info)
+ 
+ error: aborting due to 6 previous errors
---
To only update this specific test, also pass `--test-args attributes/key-value-expansion.rs`

error: 1 errors occurred comparing output.
status: exit status: 1
command: env -u RUSTC_LOG_COLOR RUSTC_ICE="0" RUST_BACKTRACE="short" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/tests/ui/attributes/key-value-expansion.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "-Z" "ignore-directory-in-diagnostics-source-blocks=/cargo" "-Z" "ignore-directory-in-diagnostics-source-blocks=/checkout/vendor" "--sysroot" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2" "--target=x86_64-unknown-linux-gnu" "--check-cfg" "cfg(test,FALSE)" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Zwrite-long-types-to-disk=no" "-Cstrip=debuginfo" "--emit" "metadata" "-C" "prefer-dynamic" "--out-dir" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/attributes/key-value-expansion" "-A" "unused" "-A" "internal_features" "-Crpath" "-Cdebuginfo=0" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/attributes/key-value-expansion/auxiliary" "--extern" "key_value_expansion=/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/attributes/key-value-expansion/auxiliary/libkey_value_expansion.so"
--- stderr -------------------------------
error: attribute value must be a literal
##[error]  --> /checkout/tests/ui/attributes/key-value-expansion.rs:21:6
   |
   |
LL | bug!((column!())); //~ ERROR attribute value must be a literal

error: attribute value must be a literal
##[error]  --> /checkout/tests/ui/attributes/key-value-expansion.rs:27:14
   |
   |
LL |         bug!("bug" + stringify!(found)); //~ ERROR attribute value must be a literal
...
...
LL | bug!();
   |
   = note: this error originates in the macro `bug` (in Nightly builds, run with -Z macro-backtrace for more info)

error: attribute value must be a literal
error: attribute value must be a literal
##[error]  --> /checkout/tests/ui/attributes/key-value-expansion.rs:48:23
   |
LL |         doc_comment! {format!("{coor}", coor = stringify!($t1)).as_str()}
...
...
LL | some_macro!(u8);
   |
   = note: this error originates in the macro `some_macro` (in Nightly builds, run with -Z macro-backtrace for more info)

error: attribute value must be a literal
error: attribute value must be a literal
##[error]  --> /checkout/tests/ui/attributes/key-value-expansion.rs:14:23
   |
LL |         #[rustc_dummy = $expr] // Any key-value attribute, not necessarily `doc`
...
...
LL | bug!((column!())); //~ ERROR attribute value must be a literal
   |
   = note: this error originates in the macro `bug` (in Nightly builds, run with -Z macro-backtrace for more info)

error: attribute value must be a literal
error: attribute value must be a literal
##[error]  --> /checkout/tests/ui/attributes/key-value-expansion.rs:30:15
   |
LL |         #[doc = $test]
...
...
LL | bug!();
   |
   = note: this error originates in the macro `bug` (in Nightly builds, run with -Z macro-backtrace for more info)

error: attribute value must be a literal
error: attribute value must be a literal
##[error]  --> /checkout/tests/ui/attributes/key-value-expansion.rs:41:15
   |
LL |         #[doc = $x]
...
...
LL | some_macro!(u8);
   |
   = note: this error originates in the macro `doc_comment` which comes from the expansion of the macro `some_macro` (in Nightly builds, run with -Z macro-backtrace for more info)

error: aborting due to 6 previous errors
error: aborting due to 6 previous errors
------------------------------------------


---- [ui] tests/ui/attributes/validation-on-associated-items-issue-121537.rs stdout ----
Saved the actual stderr to "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/attributes/validation-on-associated-items-issue-121537/validation-on-associated-items-issue-121537.stderr"
diff of stderr:

4 LL |     #[doc = MyTrait]
6 
- error: aborting due to 1 previous error
+ error: attribute value must be a literal
+   --> $DIR/validation-on-associated-items-issue-121537.rs:2:11
+   --> $DIR/validation-on-associated-items-issue-121537.rs:2:11
+    |
+ LL |     #[doc = MyTrait]
+ 
+ error: aborting due to 2 previous errors
8 
9 
9 

Note: some mismatched output was normalized before being compared
-   --> /checkout/tests/ui/attributes/validation-on-associated-items-issue-121537.rs:2:11
+ error: attribute value must be a literal
+   --> $DIR/validation-on-associated-items-issue-121537.rs:2:11
+    |
+ LL |     #[doc = MyTrait]
+ 
+ error: aborting due to 2 previous errors



The actual stderr differed from the expected stderr.
To update references, rerun the tests and pass the `--bless` flag
To only update this specific test, also pass `--test-args attributes/validation-on-associated-items-issue-121537.rs`

error: 1 errors occurred comparing output.
status: exit status: 1
command: env -u RUSTC_LOG_COLOR RUSTC_ICE="0" RUST_BACKTRACE="short" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/tests/ui/attributes/validation-on-associated-items-issue-121537.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "-Z" "ignore-directory-in-diagnostics-source-blocks=/cargo" "-Z" "ignore-directory-in-diagnostics-source-blocks=/checkout/vendor" "--sysroot" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2" "--target=x86_64-unknown-linux-gnu" "--check-cfg" "cfg(test,FALSE)" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Zwrite-long-types-to-disk=no" "-Cstrip=debuginfo" "--emit" "metadata" "-C" "prefer-dynamic" "--out-dir" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/attributes/validation-on-associated-items-issue-121537" "-A" "unused" "-A" "internal_features" "-Crpath" "-Cdebuginfo=0" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers"
--- stderr -------------------------------
error: attribute value must be a literal
##[error]  --> /checkout/tests/ui/attributes/validation-on-associated-items-issue-121537.rs:2:13
   |
   |
LL |     #[doc = MyTrait]

error: attribute value must be a literal
##[error]  --> /checkout/tests/ui/attributes/validation-on-associated-items-issue-121537.rs:2:11
   |
   |
LL |     #[doc = MyTrait]

error: aborting due to 2 previous errors
------------------------------------------



---- [ui] tests/ui/consts/issue-90878-2.rs stdout ----
Saved the actual stderr to "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/consts/issue-90878-2/issue-90878-2.stderr"
diff of stderr:

10 LL |  #![l=|x|[b;x ]]
12 
- error: aborting due to 2 previous errors
+ error: attribute value must be a literal
+   --> $DIR/issue-90878-2.rs:1:6
+   --> $DIR/issue-90878-2.rs:1:6
+    |
+ LL |  #![l=|x|[b;x ]]
+ 
+ error: aborting due to 3 previous errors
14 
15 
15 

Note: some mismatched output was normalized before being compared
-   --> /checkout/tests/ui/consts/issue-90878-2.rs:1:6
- LL |  #![l=|x|[b;x ]] //~ ERROR attribute value must be a literal
+   --> $DIR/issue-90878-2.rs:1:6
+    |
+    |
+ LL |  #![l=|x|[b;x ]]
+ 
+ error: aborting due to 3 previous errors



The actual stderr differed from the expected stderr.
To update references, rerun the tests and pass the `--bless` flag
To only update this specific test, also pass `--test-args consts/issue-90878-2.rs`

error: 1 errors occurred comparing output.
status: exit status: 1
command: env -u RUSTC_LOG_COLOR RUSTC_ICE="0" RUST_BACKTRACE="short" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/tests/ui/consts/issue-90878-2.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "-Z" "ignore-directory-in-diagnostics-source-blocks=/cargo" "-Z" "ignore-directory-in-diagnostics-source-blocks=/checkout/vendor" "--sysroot" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2" "--target=x86_64-unknown-linux-gnu" "--check-cfg" "cfg(test,FALSE)" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Zwrite-long-types-to-disk=no" "-Cstrip=debuginfo" "--emit" "metadata" "-C" "prefer-dynamic" "--out-dir" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/consts/issue-90878-2" "-A" "unused" "-A" "internal_features" "-Crpath" "-Cdebuginfo=0" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers"
--- stderr -------------------------------
error: attribute value must be a literal
##[error]  --> /checkout/tests/ui/consts/issue-90878-2.rs:1:7
   |
   |
LL |  #![l=|x|[b;x ]] //~ ERROR attribute value must be a literal

error: cannot find attribute `l` in this scope
##[error]  --> /checkout/tests/ui/consts/issue-90878-2.rs:1:5
   |
   |
LL |  #![l=|x|[b;x ]] //~ ERROR attribute value must be a literal

error: attribute value must be a literal
##[error]  --> /checkout/tests/ui/consts/issue-90878-2.rs:1:6
   |
   |
LL |  #![l=|x|[b;x ]] //~ ERROR attribute value must be a literal

error: aborting due to 3 previous errors
------------------------------------------



---- [ui] tests/ui/malformed/malformed-interpolated.rs stdout ----
Saved the actual stderr to "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/malformed/malformed-interpolated/malformed-interpolated.stderr"
diff of stderr:

18 LL | check!(0 + 0);
20 
- error: aborting due to 3 previous errors
+ error: attribute value must be a literal
+   --> $DIR/malformed-interpolated.rs:5:23
+   --> $DIR/malformed-interpolated.rs:5:23
+    |
+ LL |         #[rustc_dummy = $expr]
+ ...
+ ...
+ LL | check!(-0);
+    |
+    = note: this error originates in the macro `check` (in Nightly builds, run with -Z macro-backtrace for more info)
+ 
+ error: attribute value must be a literal
+ error: attribute value must be a literal
+   --> $DIR/malformed-interpolated.rs:5:23
+    |
+ LL |         #[rustc_dummy = $expr]
+ ...
+ ...
+ LL | check!(0 + 0);
+    |
+    = note: this error originates in the macro `check` (in Nightly builds, run with -Z macro-backtrace for more info)
+ 
+ error: aborting due to 5 previous errors
+ error: aborting due to 5 previous errors
22 
23 

Note: some mismatched output was normalized before being compared
-   --> /checkout/tests/ui/malformed/malformed-interpolated.rs:5:23
- LL | check!(-0); //~ ERROR attribute value must be a literal
-   --> /checkout/tests/ui/malformed/malformed-interpolated.rs:5:23
- LL | check!(0 + 0); //~ ERROR attribute value must be a literal
+   --> $DIR/malformed-interpolated.rs:5:23
+    |
+    |
+ LL |         #[rustc_dummy = $expr]
+ ...
+ ...
+ LL | check!(-0);
+    |
+    = note: this error originates in the macro `check` (in Nightly builds, run with -Z macro-backtrace for more info)
+ 
+ error: attribute value must be a literal
+ error: attribute value must be a literal
+   --> $DIR/malformed-interpolated.rs:5:23
+    |
+ LL |         #[rustc_dummy = $expr]
+ ...
+ ...
+ LL | check!(0 + 0);
+    |
+    = note: this error originates in the macro `check` (in Nightly builds, run with -Z macro-backtrace for more info)
+ 
+ error: aborting due to 5 previous errors
---
To only update this specific test, also pass `--test-args malformed/malformed-interpolated.rs`

error: 1 errors occurred comparing output.
status: exit status: 1
command: env -u RUSTC_LOG_COLOR RUSTC_ICE="0" RUST_BACKTRACE="short" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/tests/ui/malformed/malformed-interpolated.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "-Z" "ignore-directory-in-diagnostics-source-blocks=/cargo" "-Z" "ignore-directory-in-diagnostics-source-blocks=/checkout/vendor" "--sysroot" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2" "--target=x86_64-unknown-linux-gnu" "--check-cfg" "cfg(test,FALSE)" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Zwrite-long-types-to-disk=no" "-Cstrip=debuginfo" "--emit" "metadata" "-C" "prefer-dynamic" "--out-dir" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/malformed/malformed-interpolated" "-A" "unused" "-A" "internal_features" "-Crpath" "-Cdebuginfo=0" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers"
--- stderr -------------------------------
error: suffixed literals are not allowed in attributes
##[error]  --> /checkout/tests/ui/malformed/malformed-interpolated.rs:12:8
   |
   |
LL | check!(0u8); //~ ERROR suffixed literals are not allowed in attributes
   |
   |
   = help: instead of using a suffixed literal (`1u8`, `1.0f32`, etc.), use an unsuffixed version (`1`, `1.0`, etc.)
error: attribute value must be a literal
##[error]  --> /checkout/tests/ui/malformed/malformed-interpolated.rs:13:8
   |
   |
LL | check!(-0); //~ ERROR attribute value must be a literal

error: attribute value must be a literal
##[error]  --> /checkout/tests/ui/malformed/malformed-interpolated.rs:14:8
   |
   |
LL | check!(0 + 0); //~ ERROR attribute value must be a literal

error: attribute value must be a literal
##[error]  --> /checkout/tests/ui/malformed/malformed-interpolated.rs:5:23
   |
   |
LL |         #[rustc_dummy = $expr]
...
...
LL | check!(-0); //~ ERROR attribute value must be a literal
   |
   = note: this error originates in the macro `check` (in Nightly builds, run with -Z macro-backtrace for more info)

error: attribute value must be a literal
error: attribute value must be a literal
##[error]  --> /checkout/tests/ui/malformed/malformed-interpolated.rs:5:23
   |
LL |         #[rustc_dummy = $expr]
...
...
LL | check!(0 + 0); //~ ERROR attribute value must be a literal
   |
   = note: this error originates in the macro `check` (in Nightly builds, run with -Z macro-backtrace for more info)

error: aborting due to 5 previous errors
---
- error: aborting due to 21 previous errors; 2 warnings emitted
+ error: attribute value must be a literal
+   --> $DIR/bad-lit-suffixes.rs:30:15
+    |
+ LL | #[rustc_dummy = "string"suffix]
+ 
+ error: attribute value must be a literal
+   --> $DIR/bad-lit-suffixes.rs:34:12
+    |
+    |
+ LL | #[must_use = "string"suffix]
+ 
+ error: aborting due to 23 previous errors; 2 warnings emitted
169 
170 
170 

Note: some mismatched output was normalized before being compared
-   --> /checkout/tests/ui/parser/bad-lit-suffixes.rs:30:15
-   --> /checkout/tests/ui/parser/bad-lit-suffixes.rs:34:12
+ error: attribute value must be a literal
+   --> $DIR/bad-lit-suffixes.rs:30:15
+    |
+ LL | #[rustc_dummy = "string"suffix]
+ 
+ error: attribute value must be a literal
+   --> $DIR/bad-lit-suffixes.rs:34:12
+    |
+    |
+ LL | #[must_use = "string"suffix]
+ 
+ error: aborting due to 23 previous errors; 2 warnings emitted



The actual stderr differed from the expected stderr.
To update references, rerun the tests and pass the `--bless` flag
To only update this specific test, also pass `--test-args parser/bad-lit-suffixes.rs`

error: 1 errors occurred comparing output.
status: exit status: 1
command: env -u RUSTC_LOG_COLOR RUSTC_ICE="0" RUST_BACKTRACE="short" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/tests/ui/parser/bad-lit-suffixes.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "-Z" "ignore-directory-in-diagnostics-source-blocks=/cargo" "-Z" "ignore-directory-in-diagnostics-source-blocks=/checkout/vendor" "--sysroot" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2" "--target=x86_64-unknown-linux-gnu" "--check-cfg" "cfg(test,FALSE)" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Zwrite-long-types-to-disk=no" "-Cstrip=debuginfo" "--emit" "metadata" "-C" "prefer-dynamic" "--out-dir" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/parser/bad-lit-suffixes" "-A" "unused" "-A" "internal_features" "-Crpath" "-Cdebuginfo=0" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers"
--- stderr -------------------------------
error: suffixes on string literals are invalid
##[error]  --> /checkout/tests/ui/parser/bad-lit-suffixes.rs:4:5
   |
   |
LL |     "C"suffix //~ ERROR suffixes on string literals are invalid
   |     ^^^^^^^^^ invalid suffix `suffix`
error: suffixes on string literals are invalid
##[error]  --> /checkout/tests/ui/parser/bad-lit-suffixes.rs:8:5
   |
   |
LL |     "C"suffix //~ ERROR suffixes on string literals are invalid
   |     ^^^^^^^^^ invalid suffix `suffix`
error: suffixes on string literals are invalid
##[error]  --> /checkout/tests/ui/parser/bad-lit-suffixes.rs:30:17
   |
   |
LL | #[rustc_dummy = "string"suffix]
   |                 ^^^^^^^^^^^^^^ invalid suffix `suffix`
error: suffixes on string literals are invalid
##[error]  --> /checkout/tests/ui/parser/bad-lit-suffixes.rs:34:14
   |
   |
LL | #[must_use = "string"suffix]
   |              ^^^^^^^^^^^^^^ invalid suffix `suffix`
error: malformed `must_use` attribute input
##[error]  --> /checkout/tests/ui/parser/bad-lit-suffixes.rs:34:1
   |
   |
LL | #[must_use = "string"suffix]
   |
help: the following are the possible correct uses
   |
   |
LL - #[must_use = "string"suffix]
LL + #[must_use = "reason"]
   |
LL - #[must_use = "string"suffix]
LL + #[must_use]

error: suffixes on string literals are invalid
##[error]  --> /checkout/tests/ui/parser/bad-lit-suffixes.rs:39:15
   |
   |
LL | #[link(name = "string"suffix)]
   |               ^^^^^^^^^^^^^^ invalid suffix `suffix`
error: invalid suffix `suffix` for number literal
##[error]  --> /checkout/tests/ui/parser/bad-lit-suffixes.rs:43:41
   |
   |
LL | #[rustc_layout_scalar_valid_range_start(0suffix)]
   |                                         ^^^^^^^ invalid suffix `suffix`
   |
   = help: the suffix must be one of the numeric types (`u32`, `isize`, `f32`, etc.)
warning: extern declarations without an explicit ABI are deprecated
##[warning]  --> /checkout/tests/ui/parser/bad-lit-suffixes.rs:3:1
   |
   |
LL | extern //~ WARN missing_abi
   |
   = note: `#[warn(missing_abi)]` on by default

warning: extern declarations without an explicit ABI are deprecated
warning: extern declarations without an explicit ABI are deprecated
##[warning]  --> /checkout/tests/ui/parser/bad-lit-suffixes.rs:7:1
   |
LL | extern //~ WARN missing_abi

error: suffixes on string literals are invalid
##[error]  --> /checkout/tests/ui/parser/bad-lit-suffixes.rs:12:5
   |
   |
LL |     ""suffix; //~ ERROR suffixes on string literals are invalid
   |     ^^^^^^^^ invalid suffix `suffix`
error: suffixes on byte string literals are invalid
##[error]  --> /checkout/tests/ui/parser/bad-lit-suffixes.rs:13:5
   |
   |
LL |     b""suffix; //~ ERROR suffixes on byte string literals are invalid
   |     ^^^^^^^^^ invalid suffix `suffix`
error: suffixes on string literals are invalid
##[error]  --> /checkout/tests/ui/parser/bad-lit-suffixes.rs:14:5
   |
   |
LL |     r#""#suffix; //~ ERROR suffixes on string literals are invalid
   |     ^^^^^^^^^^^ invalid suffix `suffix`
error: suffixes on byte string literals are invalid
##[error]  --> /checkout/tests/ui/parser/bad-lit-suffixes.rs:15:5
   |
   |
LL |     br#""#suffix; //~ ERROR suffixes on byte string literals are invalid
   |     ^^^^^^^^^^^^ invalid suffix `suffix`
error: suffixes on char literals are invalid
##[error]  --> /checkout/tests/ui/parser/bad-lit-suffixes.rs:16:5
   |
   |
LL |     'a'suffix; //~ ERROR suffixes on char literals are invalid
   |     ^^^^^^^^^ invalid suffix `suffix`
error: suffixes on byte literals are invalid
##[error]  --> /checkout/tests/ui/parser/bad-lit-suffixes.rs:17:5
   |
   |
LL |     b'a'suffix; //~ ERROR suffixes on byte literals are invalid
   |     ^^^^^^^^^^ invalid suffix `suffix`

error: invalid width `1024` for integer literal
   |
   |
LL |     1234u1024; //~ ERROR invalid width `1024` for integer literal
   |
   |
   = help: valid widths are 8, 16, 32, 64 and 128

error: invalid width `1024` for integer literal
   |
   |
LL |     1234i1024; //~ ERROR invalid width `1024` for integer literal
   |
   |
   = help: valid widths are 8, 16, 32, 64 and 128

error: invalid width `1024` for float literal
   |
   |
LL |     1234f1024; //~ ERROR invalid width `1024` for float literal
   |
   = help: valid widths are 32 and 64


error: invalid width `1024` for float literal
   |
   |
LL |     1234.5f1024; //~ ERROR invalid width `1024` for float literal
   |
   = help: valid widths are 32 and 64

error: invalid suffix `suffix` for number literal
error: invalid suffix `suffix` for number literal
##[error]  --> /checkout/tests/ui/parser/bad-lit-suffixes.rs:24:5
   |
LL |     1234suffix; //~ ERROR invalid suffix `suffix` for number literal
   |     ^^^^^^^^^^ invalid suffix `suffix`
   |
   = help: the suffix must be one of the numeric types (`u32`, `isize`, `f32`, etc.)
error: invalid suffix `suffix` for number literal
##[error]  --> /checkout/tests/ui/parser/bad-lit-suffixes.rs:25:5
   |
   |
LL |     0b101suffix; //~ ERROR invalid suffix `suffix` for number literal
   |     ^^^^^^^^^^^ invalid suffix `suffix`
   |
   = help: the suffix must be one of the numeric types (`u32`, `isize`, `f32`, etc.)
error: invalid suffix `suffix` for float literal
##[error]  --> /checkout/tests/ui/parser/bad-lit-suffixes.rs:26:5
   |
   |
LL |     1.0suffix; //~ ERROR invalid suffix `suffix` for float literal
   |     ^^^^^^^^^ invalid suffix `suffix`
   = help: valid suffixes are `f32` and `f64`

error: invalid suffix `suffix` for float literal
##[error]  --> /checkout/tests/ui/parser/bad-lit-suffixes.rs:27:5
##[error]  --> /checkout/tests/ui/parser/bad-lit-suffixes.rs:27:5
   |
LL |     1.0e10suffix; //~ ERROR invalid suffix `suffix` for float literal
   |     ^^^^^^^^^^^^ invalid suffix `suffix`
   = help: valid suffixes are `f32` and `f64`

error: attribute value must be a literal
##[error]  --> /checkout/tests/ui/parser/bad-lit-suffixes.rs:30:15
##[error]  --> /checkout/tests/ui/parser/bad-lit-suffixes.rs:30:15
   |
LL | #[rustc_dummy = "string"suffix]

error: attribute value must be a literal
##[error]  --> /checkout/tests/ui/parser/bad-lit-suffixes.rs:34:12
   |
   |
LL | #[must_use = "string"suffix]

error: aborting due to 23 previous errors; 2 warnings emitted
------------------------------------------



---- [ui] tests/ui/parser/issues/issue-104620.rs stdout ----
Saved the actual stderr to "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/parser/issues/issue-104620/issue-104620.stderr"
diff of stderr:

6    |
7    = help: the suffix must be one of the numeric types (`u32`, `isize`, `f32`, etc.)
- error: aborting due to 1 previous error
+ error: attribute value must be a literal
+   --> $DIR/issue-104620.rs:3:15
+    |
+    |
+ LL | #![rustc_dummy=5z]
+ 
+ error: aborting due to 2 previous errors
10 
11 
11 

Note: some mismatched output was normalized before being compared
-   --> /checkout/tests/ui/parser/issues/issue-104620.rs:3:15
- LL | #![rustc_dummy=5z] //~ ERROR invalid suffix `z` for number literal
+   --> $DIR/issue-104620.rs:3:15
+    |
+    |
+ LL | #![rustc_dummy=5z]
+ 
+ error: aborting due to 2 previous errors



The actual stderr differed from the expected stderr.
To update references, rerun the tests and pass the `--bless` flag
To only update this specific test, also pass `--test-args parser/issues/issue-104620.rs`

error: 1 errors occurred comparing output.
status: exit status: 1
command: env -u RUSTC_LOG_COLOR RUSTC_ICE="0" RUST_BACKTRACE="short" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/tests/ui/parser/issues/issue-104620.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "-Z" "ignore-directory-in-diagnostics-source-blocks=/cargo" "-Z" "ignore-directory-in-diagnostics-source-blocks=/checkout/vendor" "--sysroot" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2" "--target=x86_64-unknown-linux-gnu" "--check-cfg" "cfg(test,FALSE)" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Zwrite-long-types-to-disk=no" "-Cstrip=debuginfo" "--emit" "metadata" "-C" "prefer-dynamic" "--out-dir" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/parser/issues/issue-104620" "-A" "unused" "-A" "internal_features" "-Crpath" "-Cdebuginfo=0" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers"
--- stderr -------------------------------
error: invalid suffix `z` for number literal
##[error]  --> /checkout/tests/ui/parser/issues/issue-104620.rs:3:16
   |
   |
LL | #![rustc_dummy=5z] //~ ERROR invalid suffix `z` for number literal
   |                ^^ invalid suffix `z`
   |
   = help: the suffix must be one of the numeric types (`u32`, `isize`, `f32`, etc.)
error: attribute value must be a literal
##[error]  --> /checkout/tests/ui/parser/issues/issue-104620.rs:3:15
   |
   |
LL | #![rustc_dummy=5z] //~ ERROR invalid suffix `z` for number literal

error: aborting due to 2 previous errors
------------------------------------------

```

</details>

--- a/compiler/rustc_attr_parsing/src/context.rs
+++ b/compiler/rustc_attr_parsing/src/context.rs
@@ -333,7 +333,10 @@ impl<'sess> AttributeParser<'sess> {
                 {
                     lit
                 } else {
-                    let guar = self.dcx().has_errors().unwrap();
+                    let guar = self.dcx().span_delayed_bug(
+                        args.span().unwrap_or(DUMMY_SP),
+                        "expr in place where literal is expected (builtin attr parsing)",
+                    );
                     ast::MetaItemLit {
                         symbol: kw::Empty,
                         suffix: None,

--- a/compiler/rustc_attr_parsing/src/parser.rs
+++ b/compiler/rustc_attr_parsing/src/parser.rs
@@ -13,7 +13,7 @@ use rustc_ast_pretty::pprust;
 use rustc_errors::DiagCtxtHandle;
 use rustc_hir::{self as hir, AttrPath};
 use rustc_span::symbol::{Ident, kw};
-use rustc_span::{DUMMY_SP, ErrorGuaranteed, Span, Symbol};
+use rustc_span::{ErrorGuaranteed, Span, Symbol};
 
 pub struct SegmentIterator<'a> {
     offset: usize,
@@ -127,7 +127,7 @@ impl<'a> ArgParser<'a> {
             }
             AttrArgs::Eq { eq_span, expr } => Self::NameValue(NameValueParser {
                 eq_span: *eq_span,
-                value: expr_to_lit(dcx, &expr),
+                value: expr_to_lit(dcx, &expr, *eq_span),
                 value_span: expr.span,
             }),
         }
@@ -348,7 +348,7 @@ impl NameValueParser {
     }
 }
 
-fn expr_to_lit(dcx: DiagCtxtHandle<'_>, expr: &Expr) -> MetaItemLit {
+fn expr_to_lit(dcx: DiagCtxtHandle<'_>, expr: &Expr, span: Span) -> MetaItemLit {
     // In valid code the value always ends up as a single literal. Otherwise, a dummy
     // literal suffices because the error is handled elsewhere.
     if let ExprKind::Lit(token_lit) = expr.kind
@@ -356,8 +356,11 @@ fn expr_to_lit(dcx: DiagCtxtHandle<'_>, expr: &Expr) -> MetaItemLit {
     {
         lit
     } else {
-        let guar = dcx.has_errors().unwrap();
-        MetaItemLit { symbol: kw::Empty, suffix: None, kind: LitKind::Err(guar), span: DUMMY_SP }
+        let guar = dcx.span_delayed_bug(
+            span,
+            "expr in place where literal is expected (builtin attr parsing)",
+        );
+        MetaItemLit { symbol: kw::Empty, suffix: None, kind: LitKind::Err(guar), span }
     }
 }
 

--- a/tests/ui/attributes/crate-type-macro-empty.rs
+++ b/tests/ui/attributes/crate-type-macro-empty.rs
@@ -1,0 +1,7 @@
+// Tests for the issue in #137589
+#[crate_type = foo!()]
+//~^ ERROR cannot find macro `foo` in this scope
+
+macro_rules! foo {} //~ ERROR unexpected end of macro invocation
+
+fn main() {}

--- a/tests/ui/attributes/crate-type-macro-empty.stderr
+++ b/tests/ui/attributes/crate-type-macro-empty.stderr
@@ -1,0 +1,20 @@
+error: unexpected end of macro invocation
+  --> $DIR/crate-type-macro-empty.rs:5:1
+   |
+LL | macro_rules! foo {}
+   | ^^^^^^^^^^^^^^^^^^^ missing tokens in macro arguments
+
+error: cannot find macro `foo` in this scope
+  --> $DIR/crate-type-macro-empty.rs:2:16
+   |
+LL | #[crate_type = foo!()]
+   |                ^^^ consider moving the definition of `foo` before this call
+   |
+note: a macro with the same name exists, but it appears later
+  --> $DIR/crate-type-macro-empty.rs:5:14
+   |
+LL | macro_rules! foo {}
+   |              ^^^
+
+error: aborting due to 2 previous errors
+

--- a/tests/ui/attributes/crate-type-macro-not-crate.rs
+++ b/tests/ui/attributes/crate-type-macro-not-crate.rs
@@ -1,0 +1,9 @@
+// Tests for the issue in #137589
+
+
+macro_rules! foo {
+    ($x:expr) => {"rlib"}
+}
+
+#[crate_type = foo!()] //~ ERROR unexpected end of macro invocation
+fn main() {}

--- a/tests/ui/attributes/crate-type-macro-not-crate.stderr
+++ b/tests/ui/attributes/crate-type-macro-not-crate.stderr
@@ -1,0 +1,17 @@
+error: unexpected end of macro invocation
+  --> $DIR/crate-type-macro-not-crate.rs:8:16
+   |
+LL | macro_rules! foo {
+   | ---------------- when calling this macro
+...
+LL | #[crate_type = foo!()]
+   |                ^^^^^^ missing tokens in macro arguments
+   |
+note: while trying to match meta-variable `$x:expr`
+  --> $DIR/crate-type-macro-not-crate.rs:5:6
+   |
+LL |     ($x:expr) => {"rlib"}
+   |      ^^^^^^^
+
+error: aborting due to 1 previous error
+

--- a/tests/ui/attributes/crate-type-macro-not-found.rs
+++ b/tests/ui/attributes/crate-type-macro-not-found.rs
@@ -1,0 +1,8 @@
+// Tests for the issue in #137589
+#[crate_type = foo!()] //~ ERROR cannot find macro `foo` in this scope
+
+macro_rules! foo {
+    ($x:expr) => {"rlib"}
+}
+
+fn main() {}

--- a/tests/ui/attributes/crate-type-macro-not-found.stderr
+++ b/tests/ui/attributes/crate-type-macro-not-found.stderr
@@ -1,0 +1,14 @@
+error: cannot find macro `foo` in this scope
+  --> $DIR/crate-type-macro-not-found.rs:2:16
+   |
+LL | #[crate_type = foo!()]
+   |                ^^^ consider moving the definition of `foo` before this call
+   |
+note: a macro with the same name exists, but it appears later
+  --> $DIR/crate-type-macro-not-found.rs:4:14
+   |
+LL | macro_rules! foo {
+   |              ^^^
+
+error: aborting due to 1 previous error
+

